records_mover/records/types.py

```diff
@@ -18,178 +20,114 @@
-    RecordsManifestEntryWithLength =\
-        Mapping[str, Union[str, bool, int, RecordsManifestEntryMetadata]]
-    LegacyRecordsManifest = Mapping[str, List[LegacyRecordsManifestEntry]]
-    RecordsManifestWithLength = Mapping[str, List[RecordsManifestEntryWithLength]]
-    UrlDetailsEntry = Dict[str, int]
-    Url = str
-    UrlDetails = Dict[Url, UrlDetailsEntry]
 
-    RecordsFormatType = str
+Url = str
+UrlDetails = Dict[Url, UrlDetailsEntry]
 
-    DelimitedVariant = str
 
-if TYPE_CHECKING:
-    HintEncoding = Literal["UTF8", "UTF16", "UTF16LE", "UTF16BE",
-                           "UTF16BOM", "UTF8BOM", "LATIN1", "CP1252"]
-else:
-    HintEncoding = str
+#
+# Note: Any expansion of these types should also be done in
+# records.jobs.hints
+#
+RecordsFormatType = Literal['delimited', 'parquet']
 
-VALID_ENCODING: List[HintEncoding] = [
-    "UTF8", "UTF16", "UTF16LE", "UTF16BE", "UTF16BOM", "UTF8BOM", "LATIN1", "CP1252"
-]
+VALID_VARIANTS = ['dumb', 'csv', 'bigquery', 'bluelabs', 'vertica']
+DelimitedVariant = Literal['dumb', 'csv', 'bigquery', 'bluelabs', 'vertica']
 
-if TYPE_CHECKING:
-    HintQuoting = Literal["all", "minimal", "nonnumeric", None]
-else:
-    HintQuoting = Optional[str]
+HintEncoding = Literal["UTF8", "UTF16", "UTF16LE", "UTF16BE",
+                       "UTF16BOM", "UTF8BOM", "LATIN1", "CP1252"]
 
-VALID_QUOTING: List[HintQuoting] = ["all", "minimal", "nonnumeric", None]
+HintQuoting = Literal["all", "minimal", "nonnumeric", None]
 
+HintEscape = Literal["\\", None]
 
-if TYPE_CHECKING:
-    HintEscape = Literal["\\", None]
-else:
-    HintEscape = Optional[str]
+# TODO: combine this and cli thingie
+HintCompression = Literal['GZIP', 'BZIP', 'LZO', None]
 
-VALID_ESCAPE: List[HintEscape] = ["\\", None]
+# The trick here works on Literal[True, False] but not on bool:
+#
+# https://github.com/python/mypy/issues/6366#issuecomment-560369716
+HintHeaderRow = Literal[True, False]
 
-# TODO: combine this and cli thingie
-if TYPE_CHECKING:
-    HintCompression = Literal['GZIP', 'BZIP', 'LZO', None]
-else:
-    HintCompression = Optional[str]
-
-VALID_COMPRESSIONS: List[HintCompression] = ['GZIP', 'BZIP', 'LZO', None]
-
-if TYPE_CHECKING:
-    # The trick here works on Literal[True, False] but not on bool:
-    #
-    # https://github.com/python/mypy/issues/6366#issuecomment-560369716
-    HintHeaderRow = Literal[True, False]
-
-    HintDoublequote = Literal[True, False]
-else:
-    HintHeaderRow = bool
-
-    HintDoublequote = bool
-
-
-if TYPE_CHECKING:
-    # TODO: This None is a bug in the spec, right?
-    HintDateFormat = Literal[None, 'YYYY-MM-DD', 'MM-DD-YYYY', 'DD-MM-YYYY', 'MM/DD/YY']
-else:
-    # TODO: This None is a bug in the spec, right?
-    HintDateFormat = Optional[str]
-
-VALID_DATEFORMATS: List[HintDateFormat] = [
-    None, 'YYYY-MM-DD', 'MM-DD-YYYY', 'DD-MM-YYYY', 'MM/DD/YY'
-]
-
-if TYPE_CHECKING:
-    HintTimeOnlyFormat = Literal["HH12:MI AM", "HH24:MI:SS"]
-else:
-    HintTimeOnlyFormat = str
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker